tests/test_build_html.py

```diff
@@ -105,14 +109,17 @@ def get_text(node):
         rex = re.compile(check)
         if be_found:
             if any(rex.search(get_text(node)) for node in nodes):
-                return
+                if limit is None:
+                    return
+                if sum(rex.search(get_text(node)) for node in nodes) <= limit:
+                    return
         else:
             if all(not rex.search(get_text(node)) for node in nodes):
                 return
 
-        raise AssertionError('%r not found in any node matching '
-                             'path %s in %s: %r' % (check, path, fname,
-                                                    [node.text for node in nodes]))
+    raise AssertionError('%r not found in any node matching '
+                         'path %s in %s: %r' % (check, path, fname,
+                                                [node.text for node in nodes]))
```

Comment:
This is a test utility method, but even so this message is now potentially misleading; it can be emitted if the `limit` check fails.

The reason I de-dented it is that a couple of the `if` branches earlier in the method _seemed_ like they should raise an exception if their checks fail.  Fortunately we did not have any true negative (test that passed despite a check failing) cases.

Reply:
It looks like I was confused here; it seems intentional that the `if callable` clause and `elif not check` clause allow logic flow to exit the method without an assertion.

In particular: for custom `check` methods -- [`tail_check`](https://github.com/sphinx-doc/sphinx/blob/35965903177c6ed9a6afb62ccd33243a746a3fc0/tests/test_build_html.py#L70-L79) -- they're responsible for raising an exception themselves on failure.

If I were to be extremely nitpicky, I'd suggest that even though the use of a [`pass` statement](https://github.com/sphinx-doc/sphinx/blob/35965903177c6ed9a6afb62ccd33243a746a3fc0/tests/test_build_html.py#L95) _could_ kinda indicate "yep, everything OK in this case, allow control-flow to continue", it's slightly inconsistent with the use of `return` statements to early-exit in other cases.  I think that's a pedantic detail though.